src/combinators/Progression.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: AGPL-v3.0
+
+pragma solidity 0.8.17;
+import "../Feedbase.sol";
+import { Ward } from '../mixin/ward.sol';
+
+contract Progression is Ward {
+    struct Config {
+        address srca;
+        bytes32 taga;
+        address srcb;
+        bytes32 tagb;
+        uint start;
+        uint end;
+        bool paused;
+    }
+
+    struct Cache {
+        uint a;
+        uint b;
+        uint fact;
+        bool end;
+        bool valid;
+    }
+
+    mapping(bytes32=>Cache) public caches;
+    mapping(bytes32=>Config) public configs;
+    Feedbase public immutable fb;
+    uint constant RAY = 10 ** 27;
+
+    constructor(Feedbase _fb) Ward() {
+        fb = _fb;
+    }
+
+    function setConfig(bytes32 tag, Config calldata _config) public _ward_ {
+        configs[tag] = _config;
+        caches[tag].fact = 0;
+    }
+
+    // smooth progression
+    function poke(bytes32 tag) public {
```

Comment:
Haven't run it but think poke doesn't look like what we want. Result should move based on relative changes of A and B. Say B is 100 times greater than A, progress is about 50%, and a goes down 10% and B goes up 10%, then the result should be about unchanged. But this implementation looks like the result will  significantly increase. If that's right then possible solution could be something like  
price = previous price * (rel change(a) * prog_share(a) + rel_change(b) * prog_share(b)  )